### ISSUE TITLE
Renegotiate Codecs on Blind Transfer

### ISSUE THREAD

**Issue by donileo** [signalwire/freeswitch#947]
When setting up a call as follows: 

A(OPUS,PCMU) <-> FS <-> B(OPUS) and then B does a Blind Transfer on A to C(PCMU) the call fails with a SIP 488 Incompatible destination even though it should not since A supports PCMU.

What occurs is that the codecs on the initial call between A and B get negotiated to be OPUS. When the blind transfer occurs FS sends an INVITE to C (which FS generates) but it only includes the OPUS codec setup in the call prior to the transfer, not the original list of codecs it sent in its invite to B. When C sees that PCMU is not supported the call fails and a SIP 488 is returned.

**Expected behavior**
Freeswitch should send the original codecs (sent by A in the invite to B) to C so that the call can succeed.

**Package version or git hash**
Freeswitch Version: `1.10.4-dev+git~20200622T145848Z~a862491bf8~64bit`

**Comment by josefig** [comment]
What you have in the dialplan? it happened to me once, and it was the problem not declared the codecs to be used in the dialplan.

**Comment by donileo** [comment]
@josefig My dialplan is as follows. I am hitting the scenario above by starting the call by doing an internal to internal call (extension 1001 to 1000) and then doing a transfer on extension 1001 to a sip URI :`904@mouselike.org`

My diaplan: 
```
<include>
  <extension name="external_sip_uri">
    <condition field="source" expression="mod_sofia"/>
    <condition field="${outside_call}" expression="^$"/>
    <condition field="${domain_exists(${sip_req_host})}" expression="false" break="on-true">
      <action application="export" data="origination_callee_id_name=${sip_to_uri}"/>
      <action application="export" data="origination_callee_id_number=${destination_number}"/>
      <action application="bridge" data="[call_timeout=35]sofia/${use_profile}/${sip_to_uri}"/>
    </condition>
    <condition field="${domain_exists(${sip_refer_to_host})}" expression="false" break="on-true">
      <action application="export" data="origination_callee_id_name=${sip_refer_to_uri}"/>
      <action application="export" data="origination_callee_id_number=${destination_number}"/>
      <!-- <action application="export" data="codec_string='${ep_codec_string}'"/> -->
      <action application="bridge" data="[call_timeout=35]sofia/${use_profile}/${sip_refer_to_uri}"/>
    </condition>
  </extension>
</include>
```

**Comment by briankwest** [comment]
You also need logs of this, I suspect this is out of order, or just wrong for dialplan.

**Comment by donileo** [comment]
@briankwest Here is a log: https://pastebin.com/S26rHzJQ
Key parts start at the REFER where you notice that even though the original SDP from client 1000 contains PCMU/PCMA the invite SDP sent to 904@mouselike.org does not contain PCMU/PCMA but instead the codec negotiated prior to the REFER.

**Comment by adil-mafzool** [comment]
Any update regarding this issue?

**Comment by shaunjstokes** [comment]
Is there any update on this?

Blind transfer fails if the device receiving the transfer does not support the negotiated codec, attended transfer is a workaround but results in transcoding.

**Comment by dragos-oancea** [comment]
likely fixed by 6c87ed491597fb5e30935d8309aa7e0c3aa9e18f . please test.
